lectures/ifp_egm.md

```diff
@@ -436,10 +435,9 @@ ifp_data = [
 ]
 
 # Use the JAX IFP data as our defaults for the Numba version
-model = ifp()
-constants, sizes, arrays = model
-β, R, γ = constants
-s_size, y_size = sizes
+model = create_household()
+β, R, γ, s_size, y_size, s_grid, y_grid, P = model
+arrays = s_grid, y_grid, P
 s_grid, y_grid, P = (np.array(a) for a in arrays)
 
 @jitclass(ifp_data)
```

Comment:
Instead of creating global variables in the cell below, I propose that we define the default values for class `IFP` using `model` directly:

```
ifp_data = [
    ('R', float64),              
    ('β', float64),             
    ('γ', float64),            
    ('P', float64[:, :]),     
    ('y_grid', float64[:]),  
    ('s_grid', float64[:])    
]

# Use the JAX IFP data as our defaults for the Numba version
model = create_household()
β, R, γ, s_size, y_size, s_grid, y_grid, P = model
arrays = s_grid, y_grid, P
s_grid, y_grid, P = (np.array(a) for a in arrays)

@jitclass(ifp_data)
class IFP:

    def __init__(self,
                 R=R,
                 β=β,
                 γ=γ,
                 P=np.array(P),
                 y_grid=np.array(y_grid),
                 s_grid=s_grid):

        self.R, self.β, self.γ = R, β, γ
        self.P, self.y_grid = P, y_grid
        self.s_grid = s_grid

        # Recall that we need R β < 1 for convergence.
        assert self.R * self.β < 1, "Stability condition violated."

    def u_prime(self, c):
        return c**(-self.γ)
    
    def u_prime_inv(self, u_prime):
        return u_prime**(-1/self.γ)
```
->
```
ifp_data = [
    ('R', float64),              
    ('β', float64),             
    ('γ', float64),   
    ('y_size', int64),
    ('s_size', int64),             
    ('P', float64[:, :]),     
    ('y_grid', float64[:]),  
    ('s_grid', float64[:]),
]

# Use the JAX IFP data as our defaults for the Numba version
model = create_household()

@jitclass(ifp_data)
class IFP:

    def __init__(self,
                 R=model.R,
                 β=model.β,
                 γ=model.γ,
                 P=np.array(model.P),
                 y_grid=np.array(model.y_grid),
                 s_grid=np.array(model.s_grid)):

        self.R, self.β, self.γ = R, β, γ
        self.P, self.y_grid = P, y_grid
        self.s_grid = s_grid
        self.y_size, self.s_size = len(y_grid), len(s_grid)

        # Recall that we need R β < 1 for convergence.
        assert self.R * self.β < 1, "Stability condition violated."

    def u_prime(self, c):
        return c**(-self.γ)
    
    def u_prime_inv(self, u_prime):
        return u_prime**(-1/self.γ)
```

Note that `y_size` will be removed from the global variable after this change. We will need to change the function `successive_approx_numba` below that is dependent on global `y_size`.

Also note that we will need to add `int64` in the import: 

```
from numba import njit, float64, int64
```

Reply:
I agree @HumphreyYang, this is neater.